physics/dynamic_frame.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <functional>
+
+#include "geometry/affine_map.hpp"
+#include "geometry/named_quantities.hpp"
+#include "geometry/orthogonal_map.hpp"
+#include "geometry/rotation.hpp"
+#include "physics/degrees_of_freedom.hpp"
+#include "quantities/quantities.hpp"
+
+namespace principia {
+
+using geometry::AffineMap;
+using geometry::AngularVelocity;
+using geometry::Instant;
+using geometry::OrthogonalMap;
+using geometry::Position;
+using geometry::Vector;
+using quantities::Acceleration;
+using quantities::Length;
+
+namespace physics {
+
+// TODO(egg): AffineMap needs an accessor for its linear map.
+template <typename FromFrame, typename ToFrame>
+using RigidTransformation =
+    AffineMap<FromFrame, ToFrame, Length, OrthogonalMap>;
+
+// The instantaneous motion of |ToFrame| with respect to |FromFrame|.
+// This is the derivative of a |RigidTransformation<FromFrame, ToFrame>|.
+// In order to invert, the |RigidTransformation| is needed, and we need its
+// linear part anyway, so we store it (and we forward its action on positions).
+template<typename FromFrame, typename ToFrame>
+class RigidMotion {
```

Comment:
I would prefer to split this file in two parts: `RigidTransformation` + `RigidMotion` in one file, `rigid_motion.hpp`.  And the interface `DynamicFrame` in `dynamic_frame.hpp`.  1 file == 1 class == good.  As written, `dynamic_frame_test.cpp` doesn't test `DynamicFrame` at all (which is fine, an interface doesn't need tests).